tests/e2e/nvidia-container-toolkit_test.go

```diff
@@ -297,4 +298,56 @@ var _ = Describe("docker", Ordered, ContinueOnFailure, func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 	})
+
+	When("Running a container where the firmware folder resolves outside the container root", Ordered, func() {
+		var outputDir string
+		BeforeAll(func(ctx context.Context) {
+			output, _, err := runner.Run("mktemp -d -p $(pwd)")
+			Expect(err).ToNot(HaveOccurred())
+			outputDir = strings.TrimSpace(output)
+
+			_, _, err = runner.Run("docker pull ubuntu")
+			Expect(err).ToNot(HaveOccurred())
+
+			_, _, err = runner.Run(`docker build -t firmware-test \
+            --build-arg RM_VERSION="$(basename $(ls -d /lib/firmware/nvidia/*.*))" \
+            --build-arg CURRENT_DIR="` + outputDir + `" \
+            - <<EOF
+FROM ubuntu
+RUN mkdir -p /lib/firmware/nvidia/
+ARG RM_VERSION
+ARG CURRENT_DIR
+RUN ln -s /../../../../../../../../\$CURRENT_DIR /lib/firmware/nvidia/\$RM_VERSION
+EOF`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func(ctx context.Context) {
+			output, _, _ := runner.Run(fmt.Sprintf("ls -A %s", outputDir))
+			Expect(output).To(BeEmpty())
+		})
+
+		AfterAll(func(ctx context.Context) {
+			if outputDir != "" {
+				runner.Run(fmt.Sprintf("rm -rf %s", outputDir))
+			}
+		})
+
+		It("should not fail when using CDI", func(ctx context.Context) {
+			output, _, err := runner.Run("docker run --rm --runtime=nvidia --gpus=runtime.nvidia.com/gpu=all firmware-test")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).To(BeEmpty())
+		})
+
+		It("should not fail when using the nvidia-container-runtime-hook", func(ctx context.Context) {
```

Comment:
This might have been a typo in my suggestion:
```suggestion
		It("should not fail when using the nvidia-container-runtime", func(ctx context.Context) {
```